src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -114,8 +114,8 @@ class ProtobufPlugin implements Plugin<Project> {
     }
 
     @TypeChecked(TypeCheckingMode.SKIP)
-    private static void linkGenerateProtoTasksToTask(Task task, GenerateProtoTask genProtoTask) {
-      task.source genProtoTask.getOutputSourceDirectorySet().include("**/*.java", "**/*.kt")
+    private static void linkGenerateProtoTasksToTask(Task task, Provider<GenerateProtoTask> genProtoTask) {
+      task.source genProtoTask.get().getOutputSourceDirectorySet().include("**/*.java", "**/*.kt")
```

Comment:
I think we can use `.map` here, like: 
`task.source(genProtoTask.map { task -> task.getOutputSourceDirectorySet().include("**/*.java", "**/*.kt") })`

Reply:
My plan is to remove all calls to getOutputSourceDirectorySet() and move that method into ProtobufPlugin so we can call it without initializing the task. But I didn't want to mix it in with this change.

For this specific location the delayed initialization adds little benefit as the compile task is being configured already (see `linkGenerateProtoTasksToTaskName()`), so the genProtoTask is likely being used. The more unfortunate case is `sourceSet.java.source(generateProtoTask.get().getOutputSourceDirectorySet())` where I had to add the `get()` to resolve the conflict.

And those plans don't require using `map{}`, which is still busted: https://github.com/gradle/gradle/issues/21179